### PR TITLE
gateway: retry connection to notification stream

### PR DESCRIPTION
**Pull request by ilackarms**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2070

### CODE REVIEW

File: projects/gateway/pkg/validation/notification_channel.go [hunk 1]
```diff
@@ -5,20 +5,22 @@ import (
 
 	"github.com/solo-io/gloo/projects/gloo/pkg/api/grpc/validation"
 	"github.com/solo-io/go-utils/contextutils"
+	"github.com/solo-io/go-utils/errors"
 	"go.uber.org/zap"
 )
 
-func MakeNotificationChannel(ctx context.Context, stream validation.ProxyValidationService_NotifyOnResyncClient) (<-chan struct{}, error) {
+func MakeNotificationChannel(ctx context.Context, client validation.ProxyValidationServiceClient) (<-chan struct{}, error) {
+	//func MakeNotificationChannel(ctx context.Context, stream validation.ProxyValidationService_NotifyOnResyncClient) (<-chan struct{}, error) {
```
Comment (GrahamGoudeau) [inline projects/gateway/pkg/validation/notification_channel.go]: remove

File: projects/gateway/pkg/validation/notification_channel.go [hunk 2]
```diff
@@ -30,7 +32,12 @@ func MakeNotificationChannel(ctx context.Context, stream validation.ProxyValidat
 
 			notification, err := stream.Recv()
 			if err != nil {
```
Comment (GrahamGoudeau) [inline projects/gateway/pkg/validation/notification_channel.go]: This is just my own ignorance about this, but does the old stream need to be explicitly closed in the case where it has errored?

Comment (ilackarms) [reply projects/gateway/pkg/validation/notification_channel.go]: according to grpc docs:
```
	// RecvMsg blocks until it receives a message into m or the stream is
	// done. It returns io.EOF when the stream completes successfully. On
	// any other error, the stream is aborted and the error contains the RPC
	// status.
```

**Comment by ilackarms** [comment]
/kick

soem e2e test code panicked, looks like a flake:
```

Step #20 - "regression-tests": 
Step #20 - "regression-tests": [AfterEach] Robustness tests
Step #20 - "regression-tests":   /workspace/gloo/test/kube2e/gateway/robustness_test.go:99
Step #20 - "regression-tests": •! Panic [93.310 seconds]
Step #20 - "regression-tests": Robustness tests [It] updates Envoy endpoints even if proxy is rejected 
Step #20 - "regression-tests": /workspace/gloo/test/kube2e/gateway/robustness_test.go:115
Step #20 - "regression-tests": 
```

**Pull request merged**